src/UI/Implementation/Render/AbstractComponentRenderer.php

```diff
@@ -85,6 +92,32 @@ public function registerResources(ResourceRegistry $registry) {
 	}
 
 	/**
+	 * Bind the component to JavaScript.
+	 *
+	 * ATTENTION: If this returns an id, the returned id has to be included as id-attribute
+	 * into the HTML of your component.
+	 *
+	 * @param	JavaScriptBindable	$component
+	 * @return	string|null
+	 */
+	final protected function bindJavaScript(JavaScriptBindable $component) {
+		$binder = $component->getOnLoadCode();
+		if ($binder === null) {
+			return null;
```

Comment:
The Idea was to only introduce what's required. Please also remember, that per design there should be no possible way to derive an id if you do not use it to bind some code.

The strange magic in the close button to me seems to be more related to the templating engine (or my lack of understanding thereof) than to the id stuff, though. If the params worked as expected, the else in maybe_render_id would be unnecessary.